### ISSUE TITLE
change config files to support override files

### ISSUE THREAD

**Issue by jakeyr** [verygood-ops/verygood.vault#3]
Currently, this role installs a config file for each application, such as `vault.http.conf.yml` into `/etc/verygood` and tells the application to use that configuration file. We are now moving to where a base configuration is packaged with the application, and then per-environment override files are installed by ansible.

So instead of the current system, we should:

  * look for a template named something like vault-{$app}.${environment}.config.yml.j2
  * if that is present (or perhaps if a variable is manually set) install that file into something like: `/etc/verygood/${app}/${environment}.yml` or `/opt/verygood/${app}/etc/${environment}.yml`
  * add `-DVAULT_{$APP}_ENV=${environment}` to the run command 
  * change the run command to use the base configuration file which is included in the tarball in the `resources` directory
  * add whatever directory contains the file to the java classpath so that the override file is found. this might need to be done in the per-application pom.xml.

**Comment by mahmoudimus** [comment]
Here's how to do it per application pom:

```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-jar-plugin</artifactId>
  <version>2.4</version>
  <configuration>
    <archive>
      <manifest>
        <addDefaultImplementationEntries>true</addDefaultImplementationEntries>
        <addClasspath>true</addClasspath>
      </manifest>
      <manifestEntries>
        <Class-Path>/etc/verygood/http/</Class-Path>
      </manifestEntries>
    </archive>
  </configuration>
</plugin>
```

**Comment by mjallday** [comment]
@mahmoudimus is it a good idea to have a info about the prod system in the pom?

**Comment by mahmoudimus** [comment]
it's not really prod system, it's just a diff class path. development or not, it is just another folder that has some resources.

**Comment by mjallday** [comment]
`java -classpath /etc/verygood/http/` works just as well right?

**Comment by mahmoudimus** [comment]
yup, it does. but you have to choose between -cp or -jar

**Comment by mjallday** [comment]
Potential implementation https://github.com/verygood-ops/verygood.vault/tree/config-override

**Comment by mjallday** [comment]
Blocking on https://github.com/verygoodgroup/vault-http/issues/23

**Comment by mahmoudimus** [comment]
Fixed via #5
